common.js

```diff
@@ -0,0 +1,142 @@
+let fs = require('fs');
+const path = require('path');
+
+exports.delete_env = async function (client, env_name, nmspace) {
+
+    const del_env = await client.apis['fission.io'].v1.namespaces(nmspace).environments(env_name).delete();
+    console.log("Fission envirnment ",del_env['body']['details']['name']," deleted");
+}
+exports.delete_fn = async function (client, fn_name, nmspace) {
+    const del_fn = await client.apis['fission.io'].v1.namespaces(nmspace).functions(fn_name).delete();
+    console.log("Fission function ",del_fn['body']['details']['name']," deleted");
+}
+exports.fn_code = async function (client, fn_name, nmspace) {
+
+    const fn_data = await client.apis['fission.io'].v1.namespaces(nmspace).functions(fn_name).get();
+    const pkg_name = fn_data['body']['spec']['package']['packageref']['name'];
+    const pkg_namespace = fn_data['body']['spec']['package']['packageref']['namespace'];
+    const pkg_data = await client.apis['fission.io'].v1.namespaces(pkg_namespace).packages(pkg_name).get();
+    const pkg_code = pkg_data['body']['spec']['deployment']['literal'];
+    const pkg_decode = new Buffer(pkg_code, 'base64');
+    console.log(pkg_decode.toString());
+}
+exports.fn_info = async function (client, fn_name, nmspace) {
+    const fn_info = await client.apis['fission.io'].v1.namespaces(nmspace).functions(fn_name).getStream();
+    console.log(fn_info);
+}
+
+ exports.create_env = async function (client,env_name,nmspace,img) {
+    const env = {
+        apiVersion: 'fission.io/v1',
+        kind: 'Environment',
+        metadata: {
+            name: env_name,
+            namespace: nmspace,
+        },
+        spec: {
+            TerminationGracePeriod: 360,
+            builder: {},
+            keeparchive: false,
+            poolsize: 1,
+            resources: {},
+            runtime: {
+                functionendpointport: 0,
+                image: img,
+                loadendpointpath: '',
+                loadendpointport: 0
+            },
+        version: 1
+        }
+    }
+    const envs = await client.apis['fission.io'].v1.namespaces('default').environments.post({body: env});
+    console.log("Fission environment ",envs['body']['metadata']['name']," created");
+}
+
+async function create_func(client,funcname,env_name,pkg_name) {
+
+    const fn = {
+        apiVersion: 'fission.io/v1',
+        kind: 'Function',
+        metadata: {
+            clusterName: '',
+            name: funcname,
+            namespace: 'default',
+        },
+        spec: {
+            InvokeStrategy: {
+                ExecutionStrategy: ['poolmgr'],
+                StrategyType: 'execution'
+            },
+            configmaps: null,
+            environment: {
+                name: env_name,
+                namespace: 'default',
+            },
+            package: {
+                packageref: {
+                    namespace: 'default',
+                    name: pkg_name['metadata']['name'],
```

Comment:
same here you have used pkg_name['metadata']['name'], it will give error if any of the key is not present.

Reply:
we are calling a fission function's package. Every function in fission by default must have a package

Reply:
ok